workflows/offline_ml_diags/offline_ml_diags/_helpers.py

```diff
@@ -46,6 +48,23 @@
 ScalarMetrics = Dict[str, Mapping[str, float]]
 
 
+def get_variable_indices(
+    data: xr.Dataset, variables: Sequence[str]
+) -> Mapping[str, Tuple[int, int]]:
+    if "time" in data.dims:
+        data = data.isel(time=0).squeeze(drop=True)
+    stacked = data.stack(sample=["tile", "x", "y"])
+    variable_dims = count_features_2d(
+        variables, stacked.transpose("sample", ...), "sample"
+    )
+    start = 0
+    variable_indices = {}
+    for var in variables:
+        variable_indices[var] = (start, start + variable_dims[var])
+        start += variable_dims[var]
```

Comment:
I assume variable_dims retains the same order as variables so you could use the `.items()` instead of two dictionary accesses.